jerry-core/api/jerry.c

```diff
@@ -3088,6 +3089,86 @@ jerry_typedarray_find_by_type (jerry_typedarray_type_t type_name, /**< type of t
 
 #endif /* !CONFIG_DISABLE_ES2015_TYPEDARRAY_BUILTIN */
 
+/**
+ * Calculates the size of the given pattern and creates a RegExp object.
+ *
+ * @return value of the Constructed RegExp object.
+ */
+jerry_value_t
+jerry_create_regexp (const jerry_char_t *pattern,
+                     jerry_regexp_flags_t flags)
+{
+  return jerry_create_regexp_sz (pattern, lit_zt_utf8_string_size (pattern), flags);
+} /* jerry_create_regexp */
+
+/**
+ * Creates a RegExp object with the given pattern and flags.
+ *
+ * @return value of the Constructed RegExp object.
+ */
+jerry_value_t
+jerry_create_regexp_sz (const jerry_char_t *pattern, /**< RegExp pattern */
+                        jerry_size_t pattern_size, /**< Length of the pattern */
+                        jerry_regexp_flags_t flags) /**< Optional RegExp flags */
+{
+  jerry_assert_api_available ();
+
+#ifndef CONFIG_DISABLE_REGEXP_BUILTIN
+  if (!lit_is_valid_utf8_string (pattern, pattern_size))
+  {
+    return jerry_throw (ecma_raise_common_error ("Input must be a valid utf8 string"));
+  }
+
+  ecma_string_t *ecma_pattern = ecma_new_ecma_string_from_utf8 (pattern, pattern_size);
+
+  ecma_value_t str_val = ecma_make_string_value (ecma_pattern);
+
+  jerry_value_t ret_val;
+
+  if (flags > 0)
+  {
+    ecma_value_t flag_val = ECMA_VALUE_UNDEFINED;
+    jerry_char_t flag_str[3];
+    unsigned char *str_ptr = flag_str;
+
+    if (flags & JERRY_REGEXP_FLAG_GLOBAL)
+    {
+      *str_ptr++ = 'g';
+    }
+
+    if (flags & JERRY_REGEXP_FLAG_MULTILINE)
+    {
+      *str_ptr++ = 'm';
+    }
+
+    if (flags & JERRY_REGEXP_FLAG_IGNORE_CASE)
+    {
+      *str_ptr++ = 'i';
+    }
+
+    flag_val = jerry_create_string_sz (flag_str, (jerry_size_t) (str_ptr - flag_str));
+
+    ecma_value_t args[2] = {str_val, flag_val};
+    ret_val = ecma_builtin_regexp_dispatch_construct (args, 2);
+    jerry_release_value (flag_val);
+  }
+  else
+  {
+    ret_val = ecma_builtin_regexp_dispatch_construct (&str_val, 1);
+  }
+
+  ecma_free_value (str_val);
+  return ret_val;
+
+#else /* CONFIG_DISABLE_REGEXP_BUILTIN */
+  JERRY_UNUSED (pattern);
+  JERRY_UNUSED (pattern_size);
+  JERRY_UNUSED (flags);
+
+  return jerry_throw (ecma_raise_type_error (ECMA_ERR_MSG ("RegExp is not supported.")));
+#endif /* !CONFIG_DISABLE_REGEXP_BUILTIN */
+} /* jerry_create_regexp_sz */
```

Comment:
Strange placement of these functions. They are surrounded by TypedArray-related function.